### ISSUE TITLE
Lower Recall in Enhanced YOLOv8 (CBAM + MBConv) vs Base

### ISSUE THREAD

**Issue by KeyzCoder** [ultralytics/ultralytics#22020]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

<html>
<body>
<p data-start="133" data-end="138">Hi,</p>
<p data-start="140" data-end="263">I trained both my base YOLOv8 model and an enhanced version (YOLOv8 + CBAM + MBConv) for 300 epochs using these settings:</p>
<pre class="overflow-visible!" data-start="265" data-end="364"><div class="contain-inline-size rounded-2xl relative bg-token-sidebar-surface-primary"><div class="sticky top-9"><div class="absolute end-0 bottom-0 flex h-9 items-center pe-2"><div class="bg-token-bg-elevated-secondary text-token-text-secondary flex items-center gap-4 rounded-sm px-2 font-sans text-xs"></div></div></div><div class="overflow-y-auto p-4" dir="ltr"><code class="whitespace-pre!"><span><span><span class="hljs-attr">epochs</span></span><span>=</span><span><span class="hljs-number">300</span></span><span> imgsz=</span><span><span class="hljs-number">640</span></span><span> batch=-</span><span><span class="hljs-number">1</span></span><span> lr0=</span><span><span class="hljs-number">0.005</span></span><span> warmup_epochs=</span><span><span class="hljs-number">5</span></span><span> lrf=</span><span><span class="hljs-number">0.01</span></span><span> close_mosaic=</span><span><span class="hljs-number">10</span></span><span> plots=</span><span><span class="hljs-literal">True</span></span><span>
</span></span></code></div></div></pre>
<p data-start="366" data-end="389">Here are the results:</p>

Here are the results of my training (300 epochs):

| Model          | Precision | Recall  | mAP50   | mAP50-95 |
|----------------|-----------|---------|---------|----------|
| Base           | 0.5556    | 0.5253  | 0.5056  | 0.2092   |
| Enhanced       | 0.6253    | 0.4957  | 0.5149  | 0.2051   |
| MBConv Only    | 0.5904    | 0.5216  | 0.5014  | 0.2112   |
| CBAM Only      | 0.5924    | 0.5168  | 0.5069  | 0.2189   |

<p data-start="764" data-end="787">I noticed two things:</p>
<ol data-start="788" data-end="1123">
<li data-start="788" data-end="941">
<p data-start="791" data-end="941">The <strong data-start="795" data-end="851">Enhanced model has better precision but lower recall</strong> compared to the base. Why might this be happening, and what should I try to balance it?</p>
</li>
<li data-start="942" data-end="1123">
<p data-start="945" data-end="1123">The overall results are still not reaching 70–90% (as seen in other YOLO papers). Could this be mostly are my hyperparameters limiting the performance or something that I may have missed?</p>
</li>
</ol>
<p data-start="1125" data-end="1221">Any advice on improving recall and overall mAP for the enhanced model would be really helpful.</p>
<p data-start="1223" data-end="1232">Thanks!</p>
</body>
</html>

**Comment by UltralyticsAssistant** [comment]
👋 Hello @KeyzCoder, thanks for opening this issue and for sharing your results 🚀 This is an automated response to help get things moving quickly—an Ultralytics engineer will also assist you here shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible to help us assess your setup:
-Exact training command or Python code used
-Model YAMLs for both base and enhanced architectures (CBAM/MBConv changes)
-Training logs and plots (PR curves, confusion matrix, results.png, metrics.csv)
-Dataset details: size, classes, sample images, class balance, and any data splits
-Augmentation/hyperparameters used (you shared some—please include full config if possible)
-Environment details: OS, GPU(s), CUDA, Python, PyTorch, and `ultralytics` version (`python -c "import ultralytics, torch; print(ultralytics.__version__, torch.__version__)"`)
-Any seeds, resume/finetune settings, and checkpoint sources

Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) ✅

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Once you share the above details, we can better understand the precision/recall behavior you’re observing and provide targeted guidance 📈

**Comment by glenn-jocher** [comment]
Thanks for the clear comparison—what you’re seeing is the expected precision–recall trade-off; your CBAM/MBConv variant is likely more selective, lifting precision while suppressing low‑confidence positives, so tune the operating point using the saved PR/F1 curves and compare metrics carefully for mAP50 vs mAP50‑95. See the precision–recall discussion and curve guidance in the Ultralytics resources: [precision–recall trade‑off](https://www.ultralytics.com/blog/accuracy-precision-recall) and the [YOLO performance metrics guide](https://docs.ultralytics.com/guides/yolo-performance-metrics/).

For recall recovery at inference, try a lower confidence and slightly higher NMS IoU:
```bash
yolo predict model=enhanced.pt source=path/to/val/images conf=0.15 iou=0.7
```
For training, consider retuning hparams and scale: larger `imgsz` and multi-scale often help, especially if objects are small, and tiling can help too; details in [hyperparameter tuning](https://docs.ultralytics.com/guides/hyperparameter-tuning/) and [evaluation & fine‑tuning insights](https://docs.ultralytics.com/guides/model-evaluation-insights/).
```bash
yolo detect train model=yolo11s.pt data=your.yaml imgsz=960 epochs=300 multi_scale=True lr0=0.01 plots=True
```
Also, many papers cite mAP@0.5; your numbers (mAP50 ≈ 0.51 vs mAP50‑95 ≈ 0.21) are consistent with that gap—use the PR/F1 curves to pick an operating `conf` that balances your costs. If you can share `PR_curve.png`, `F1_curve.png`, and the confusion matrix from your run directory, we can take a quick look.

**Comment by KeyzCoder** [comment]
Thanks for the detailed explanation! I'll try your suggestions on confidence/NMS and the training tweaks. I'll also attach the PR/F1 curves and confusion matrices from both the base and enhanced runs for comparison.
Base Model:
<img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/fc77aca2-2350-4bdf-ad6b-6308dc3b7dd7" /><img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/458c3175-568a-4ab4-8d1c-b77688bb0c2f" />
<img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/f810b8df-5325-4fec-8842-6f23c77d8044" /><img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/4e2363c2-ba37-4412-9738-4ba86f35c8d4" />

Enhanced model:

<img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/3ec26d43-0b99-4735-9839-a98b822db5bf" /><img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/d689f40e-7b44-46e4-80b8-071a8dd1b079" />
<img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/3cab457f-c259-426a-b971-f57a6841fd25" /><img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/48391d30-0cc3-4e45-9a5f-a00fa4ccec4a" />

**Comment by glenn-jocher** [comment]
Thanks for the plots—your enhanced PR curve looks more selective (higher P at lower R), so pick the operating threshold empirically and, if classes overlap, try class‑agnostic NMS; a quick sweep that prints P/R/mAP across `conf` helps choose the best point:

```python
from ultralytics import YOLO

m = YOLO("enhanced.pt")
for c in [0.05, 0.1, 0.15, 0.2, 0.25]:
    r = m.val(data="your.yaml", conf=c, iou=0.65, plots=False)
    print(f"conf={c:.2f}  P={r.box.mp:.3f}  R={r.box.mr:.3f}  mAP50={r.box.map50:.3f}  mAP50-95={r.box.map:.3f}")
```

For inference with overlapping classes, test:
```bash
yolo val model=enhanced.pt data=your.yaml conf=0.12 iou=0.65 agnostic_nms=True
```

If you update any labels, delete `labels/*.cache` and re‑val to ensure metrics reflect fixes. Background on why recall drops as selectivity rises is in the Ultralytics [Recall glossary](https://www.ultralytics.com/glossary/recall), and the F1/PR thresholding used in validation is detailed in the [metrics reference](https://docs.ultralytics.com/reference/utils/metrics/).
